src/selectField/SelectField.tsx

```diff
@@ -47,13 +47,22 @@ export const SelectField = React.forwardRef(
     const a11yAttrs = pickA11yProps<SelectFieldProps>(props)
     const baseClassName = 'kirk-selectField'
     const [hasFocus, setFocus] = useState(false)
+    const [previousDefaultValue, setPreviousDefaultValue] = useState(defaultValue)
+    const [currentValue, setCurrentValue] = useState(defaultValue)
 
     useEffect(() => {
       if (ref && !disabled && focus) {
         ref.current.focus()
       }
     }, [disabled, focus])
 
+    useEffect(() => {
+      if (defaultValue !== previousDefaultValue) {
+        setPreviousDefaultValue(defaultValue)
```

Comment:
Do we need to have previous default value? Why don't we compare with  `props.defaultValue` with  `currentValue`?

Reply:
If we update the value (by using the form), we might end up overriding it i guess. Except that the useEffect dependencies will prevent this?

Reply:
I'm not sure about your question. But how I understand the code is that we do
```
if (defaultValue !== previousDefaultValue) {
```
but since `previousDefaultValue` and `currentValue` hold the same value I don't see the difference with this:
```
if (defaultValue !== currentValue) {
```

Reply:
Except that `currentValue` is updated on change

Reply:
Thanks that was the bit I was missing 🤦